var/spack/repos/builtin/packages/py-smartsim/package.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class PySmartsim(PythonPackage):
+    """A workflow library to easily integrate machine learning libraries with
+    high performance computing simulations and applications
+    """
+
+    homepage = "https://www.craylabs.org/docs/overview.html"
+    git = "https://github.com/CrayLabs/SmartSim"
+    pypi = "smartsim/smartsim-0.5.0.tar.gz"
+
+    maintainers("MattToast")
+
+    version("0.5.0", sha256="35b36243dc84af62261a7f772bae92f0b3502faf01401423899cb2a48339858c")
+
+    variant("torch", default=True, description="Build with the pytorch backend")
+
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-setuptools", type=("build",))
+    depends_on("py-cmake@3.13:", type=("build",))
+
+    depends_on("py-psutil@5.7.2:", type=("build", "run"))
+    depends_on("py-coloredlogs@10:", type=("build", "run"))
+    depends_on("py-tabulate@0.8.9:", type=("build", "run"))
+    depends_on("py-redis@4.5:", type=("build", "run"))
+    depends_on("py-tqdm@4.50.2:", type=("build", "run"))
+    depends_on("py-filelock@3.4.2:", type=("build", "run"))
+    depends_on("py-protobuf@3.20:3", type=("build", "run"))
+
+    # Companion libs
+    depends_on("py-smartredis@0.4.1", type=("build", "run"), when="@0.5.0")
+
+    # Backends
+    depends_on("redis@7.0.5:", type=("build", "run"))
+    depends_on("redis-ai@1.2.7:", type=("build", "run"))
```

Comment:
Where do these come from?

Reply:
SmartSim defines sensible and well tested lower bounds for these dependencies for the CLI to fetch in the `smartsim._core._install.buildenv.Versioner` class, but allows users to upgrade explicitly by setting environment variables.

Will add a comment pointing users to that class!